include/flatbuffers/stl_emulation.h

```diff
@@ -26,16 +26,17 @@
 #include <memory>
 #include <limits>
 
+#ifndef FLATBUFFERS_USE_STD_OPTIONAL
+  #define FLATBUFFERS_USE_STD_OPTIONAL 1
```

Comment:
Ah, this part should change from `1` to `(defined(__cplusplus) && __cplusplus >= 201703L) \
    || (defined(_MSVC_LANG) &&  _MSVC_LANG >= 201703L))`. We should only enable the flag if the compiler supports it.

My bad for suggesting the wrong fix!

Reply:
That's alright. I appreciate you sticking with this and diagnosing the build errors that emerge; I was non-plussed by them.